### PR TITLE
[feat][staking] add beacon chain staking api

### PR DESCRIPTION
**Pull request by francislyj**

### CODE REVIEW

File: src/main/java/com/binance/dex/api/client/domain/stake/beaconchain/BeaconChainDelegation.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    public BeaconChainDelegation() {
+    }
+
+    public BeaconChainDelegation(Delegation delegation, Token balance) {
+        this.delegation = delegation;
+        this.balance = balance;
+    }
+
+    public static BeaconChainDelegation createBySideChainDelegation(SideChainDelegation sideChainDelegation) {
```
Comment (owen-reorg) [inline src/main/java/com/binance/dex/api/client/domain/stake/beaconchain/BeaconChainDelegation.java:24]: It's weird to create beacon chain delegation by sidechain delegation.
It seems the reason is to reuse the sidechainDelegation related code.
Is it possible to refactor the code with a single `Delegation` class used by both beacon chain and sidechain, while remaining backward compatibility?

Comment (francislyj) [reply src/main/java/com/binance/dex/api/client/domain/stake/beaconchain/BeaconChainDelegation.java]: Common class names belong to side chain and beacon chain have been refactored.

**Review by forcodedancing — APPROVED** [review]

File: src/main/java/com/binance/dex/api/client/domain/stake/Delegation.java [hunk 2]
```diff
@@ -1,29 +1,29 @@
-package com.binance.dex.api.client.domain.stake.sidechain;
+package com.binance.dex.api.client.domain.stake;
 
 import com.binance.dex.api.client.encoding.message.Token;
 
 /**
  * @author Fitz.Lu
  **/
-public class SideChainDelegation {
```
Comment (forcodedancing) [inline src/main/java/com/binance/dex/api/client/domain/stake/Delegation.java:8]: Is the renaming of class names and package names necessary?
We need to point this change out in release, to notify our users.

Comment (forcodedancing) [reply src/main/java/com/binance/dex/api/client/domain/stake/Delegation.java:8]: ok, we just copy these files.

**Review by forcodedancing — CHANGES_REQUESTED** [review]
Can we also refine the description of this pr?

**Review by forcodedancing — CHANGES_REQUESTED** [review]
It is better we can change `pom.xml` to upgrade the sdk to a newer version also.

**Review by forcodedancing — CHANGES_REQUESTED** [review]
https://github.com/bnb-chain/java-sdk/blob/master/src/main/java/com/binance/dex/api/client/TransactionConverter.java#L538 Also need to implement transaction converters.

**Pull request merged**